### PR TITLE
chore: integration test all three major usage modes

### PR DESCRIPTION
**Pull request by RomainMuller**
Integration test:
- `orchestrion go build`
- `go build -toolexec 'orchestrion toolexec'`
- `CGOFLAGS="'-toolexec=orchestrion toolexec'" go build`

### CODE REVIEW

File: internal/injector/injector.go [hunk 1]
```diff
@@ -62,17 +63,28 @@ type (
...
 			packages.NeedTypes |
 			packages.NeedTypesSizes |
 			packages.NeedSyntax |
 			packages.NeedTypesInfo,
 	}
+	if flags, err := goflags.Flags(); err == nil {
+		// Honor any `-tags`  flags provided by the user, as these may affect what
+		// is getting compiled or not.
+		if tags, hasTags := flags.Get("-tags"); hasTags {
+			cfg.BuildFlags = append(cfg.BuildFlags, fmt.Sprintf("-tags=%s", tags))
```
Comment (Julio-Guerra) [inline internal/injector/injector.go:84]: just curious about this: did you uncover this case some of your integration tests?

Comment (RomainMuller) [reply internal/injector/injector.go:84]: No, but I was thinking about how to design a scenario that'd be a straight-forward demonstration of this.

**Review by Hellzy — APPROVED** [review]
LGTM

File: integration-tests.ps1 [hunk 2]
```diff
@@ -147,16 +147,41 @@ for ($i = 0 ; $i -lt $tests.Length ; $i++)
...
       $null = New-Item -ItemType Directory -Path $env:GOTMPDIR # The directory must exist...
-      & $orchestrion go -C $integ build -work -o $bin "./tests/$($name)" 2>&1 1>(Join-Path $outDir "build.log")
+      switch ($env:TESTCASE_BUILD_MODE)
+      {
+        "TOOLEXEC"
+        {
+          Write-Output "[$($name)]: Building with manual -toolexec command"
+          go -C $integ build `
+            -toolexec "$($orchestrion) toolexec" `
+            -work -o $bin "./tests/$($name)" 2>&1 1>(Join-Path $outDir "build.log")
```
[the remark this answers is not in the document]
Comment (RomainMuller) [reply integration-tests.ps1:160]: Yeah I'll be getting rid of this powershell whenever I get the chance.

**Pull request merged**